plugins/GSdx/Window/GSWndEGL.cpp

```diff
@@ -41,19 +41,24 @@ int GSWndEGL::SelectPlatform()
 		return 0;
 	}
 
+	// Inspect the provided display handle to check if it is Wayland or X11.
+	// The Wayland GS display handle happens to set pDsp[1] to NULL, so check it.
+	bool is_wayland_display = display_handle == nullptr
+		|| ((void **)display_handle)[1] == nullptr;
```

Comment:
Why not use a struct there? Would be more appropriate

Reply:
I used a struct for this in `PAD` but didn't want to duplicate the code into GSdx since it'll be merged. After the GSdx merge a ton of indirection can be removed since we won't need to marshal window handles through the `pDsp` pointer anymore.

Reply:
You can always duplicate the struct for now and I'll remove it when I'll do the merge, this way I don't end up forgetting about it and casting structures